src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -550,6 +550,12 @@ where
         record_sql_in_span(&sql);
         let session = self.session.clone().unwrap();
 
+        if session.check_idle_in_transaction_timeout() {
+            self.process_terminate();
+            return Err(PsqlError::SimpleQueryError(
```

Comment:
What about adding a new error variant? I believe the handling logic can be quite similar to handling `Panic` here.

https://github.com/risingwavelabs/risingwave/blob/240416f43bae35c11d1d8a6d9e081fdd42fcc6bb/src/utils/pgwire/src/pg_protocol.rs#L350-L360